File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/file/FileConfig.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.file;

import java.io.File;
import java.io.IOException;

public abstract class FileConfig {

    private final File file;

    /**
     * Constructor of config
     *
     * @param file of config
     */
    public FileConfig(final File file) {
        this.file = file;
    }

    /**
     * Load config from file
     *
     * @throws IOException
     */
    protected abstract void loadConfig() throws IOException;

    /**
     * Save config to file
     *
     * @throws IOException
     */
    protected abstract void saveConfig() throws IOException;

    /**
     * Create config
     *
     * @throws IOException
     */
    public void createConfig() throws IOException {
        file.createNewFile();
    }

    /**
     * @return config file exist
     */
    public boolean hasConfig() {
        return file.exists();
    }

    /**
     * @return file of config
     */
    public File getFile() {
        return file;
    }
}

File: src/main/java/net/ccbluex/liquidbounce/file/FileConfig.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.file

import java.io.File
import java.io.IOException

abstract class FileConfig(val file: File) {

    /**
     * Load config from file
     *
     * @throws IOException
     */
    @Throws(IOException::class)
    abstract fun loadConfig()

    /**
     * Save config to file
     *
     * @throws IOException
     */
    @Throws(IOException::class)
    abstract fun saveConfig()

    /**
     * Create config
     *
     * @throws IOException
     */
    @Throws(IOException::class)
    fun createConfig() = file.createNewFile()

    /**
     * @return config file exist
     */
    fun hasConfig() = file.exists() && file.length() > 0

    /**
     * Load defaults when config file doesn't exist.
     */
    open fun loadDefault() {}
}